charts/hami/templates/device-plugin/daemonsetvalidator-configmap.yaml

```diff
@@ -0,0 +1,48 @@
+{{- if .Values.devicePlugin.autoValidatorDriver }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: driver-validator-configmap
+data:
+  validator.sh: |
+    #!/bin/zsh
+    if [ -f "/host/usr/bin/nvidia-smi" ]; then
+        driverRoot="/host"
+        isHostDriver="true"
+    else
+        driverRoot="/run/nvidia/driver"
+        isHostDriver="false"
+    fi
+
+    # check this directory /run/nvidia/driver does exist?
+    if [ "$isHostDriver" = "false" ]; then
+      if [ ! -d "/run/nvidia/driver" ]; then
+        echo current node $NODE_NAME not GPU device; sleep infinity
+      fi
+    fi
+
+    sleep_interval=5
+    if [ "$isHostDriver" = "false" ]; then
+        while true; do
+            stat /run/nvidia/validations/.driver-ctr-ready
+            if [ $? -ne 0 ]; then
+              kubectl label node $NODE_NAME hami.io/nvidia-selector="off" --overwrite
+              sleep $sleep_interval
+            else
+              break
+            fi
+        done
+    fi
+    while true; do
+        chroot "$driverRoot" nvidia-smi
+        if [ $? -eq 0 ]; then
+            sleep 30
+            kubectl label node $NODE_NAME hami.io/nvidia-selector="on" --overwrite
+            break
+        else
+            sleep $sleep_interval
+        fi
+    done
+    echo validations are successful; sleep infinity
```

Comment:
Once entering `sleep infinity`, regardless of any changes to the `NVIDIA driver`, this shell script will not execute any logic.

Does this mean that this shell only has one chance to execute?

Reply:
Yes, Currently, this function can solve the problem of verifying whether there is an available driver on the node after a new node or the node is restarted. If the driver is uninstalled during operation, this scenario is not solved. We can record an issue to follow up on whether this problem exists.